Raven.Client.Lightweight/Document/BulkInsertOperation.cs

```diff
@@ -105,6 +107,8 @@ public void Store(object entity, string id)
 
         public void Store(RavenJObject document, RavenJObject metadata, string id, int? dataSize = null)
         {
+            id = string.IsNullOrWhiteSpace(id) ? Guid.NewGuid().ToString() : id.Trim();
```

Comment:
Idan, Did you also update place where space was appended to id? I assume this issue was created during some experiments and you noticed you have id with spaces in handler.